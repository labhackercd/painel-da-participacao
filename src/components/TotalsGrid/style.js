/* eslint-disable import/prefer-default-export */
import { makeStyles } from '@material-ui/core/styles';

export const useStyles = makeStyles((theme) => ({
  root: {
    display: 'flex',
    flexWrap: 'wrap',
    flex: 1,
    flexDirection: 'row',
    backgroundColor: '#000000',
    justifyContent: 'space-between',
    overflow: 'hidden',
    left: 0,
  },
  text: {
    color: theme.palette.white.main,
    alignItems: 'center',
    padding: '4px 8px 0px 16px',
  },
  info: {
    color: '#FFF',
    alignSelf: 'center',
    fontWeight: 'bold',
    fontSize: '20px',
    paddingLeft: '16px',
  },
}));
