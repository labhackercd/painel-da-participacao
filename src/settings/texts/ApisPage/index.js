/* eslint-disable import/prefer-default-export */

export const apiCardsInfo = [
  {
    title: 'e-Democracia',
    infoText: 'A plataforma e-Democracia é um portal que reúne ferramentas de participação online no processo legislativo da Câmara dos Deputados. Acesse os dados e contribua para o código-fonte.',
    toolColor: '#1181E9',
    urls: {
      github: `${process.env.NEXT_PUBLIC_EDEMOCRACIA_GITHUB_URL}`,
      api: `${process.env.NEXT_PUBLIC_EDEMOCRACIA_SWAGGER_URL}`,
      site: `${process.env.NEXT_PUBLIC_EDEMOCRACIA_BASE_URL}`,
    },
  },
  {
    title: 'Audiências Interativas',
    infoText: 'A ferramenta permite a participação em audiências públicas - e outros tipos de eventos - por meio do envio de perguntas e mensagens.',
    toolColor: '#F59D2A',
    urls: {
      github: `${process.env.NEXT_PUBLIC_AUDIENCIAS_GITHUB_URL}`,
      api: `${process.env.NEXT_PUBLIC_AUDIENCIAS_SWAGGER_URL}`,
      site: `${process.env.NEXT_PUBLIC_AUDIENCIAS_SITE_URL}`,
    },
  },
  {
    title: 'Wikilegis',
    infoText: 'Disponibiliza propostas legislativas para análise dos cidadãos que opinam  em trechos do texto e avaliam as opiniões uns dos outros.',
    toolColor: '#14D768',
    urls: {
      github: `${process.env.NEXT_PUBLIC_WIKILEGIS_GITHUB_URL}`,
      api: `${process.env.NEXT_PUBLIC_WIKILEGIS_SWAGGER_URL}`,
      site: `${process.env.NEXT_PUBLIC_WIKILEGIS_SITE_URL}`,
    },
  },
  {
    title: 'E-Monitor',
    infoText: 'Mostra o histórico de utilização de ferramentas na plataforma e-Democracia por meio de gráficos, totais e tabelas.',
    toolColor: '#E438B4',
    urls: {
      github: `${process.env.NEXT_PUBLIC_PARTICIPATION_GITHUB_URL}`,
      api: null,
      site: `${process.env.NEXT_PUBLIC_PARTICIPATION_SITE_URL}`,
    },
  },
];

export const moreCardsInfo = [
  {
    title: 'eDemocracia aberto',
    infoText: 'Disponibiliza as ferramentas do e-Democracia com guias para a instalação, administração e utilização. Cada organização pode escolher quais ferramentas baixar e adotar.',
    toolColor: '#CE242E',
    urls: {
      github: null,
      api: null,
      site: `${process.env.NEXT_PUBLIC_EDEMOCRACIA_ABERTO_SITE_URL}`,
    },
  },
  {
    title: 'Portal de Dados Abertos',
    infoText: 'Na página você encontra um conjunto amplo de dados públicos sobre a Câmara dos Deputados e os trabalhos desenvolvidos. Há dados sobre votações, deputados, legislaturas, etc.',
    toolColor: '#7F69EE',
    urls: {
      github: null,
      api: null,
      site: `${process.env.NEXT_PUBLIC_DADOS_ABERTOS_SITE_URL}`,
    },
  },
];
