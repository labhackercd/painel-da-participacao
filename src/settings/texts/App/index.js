// Page SEO (Google)
export const applicationName = 'Painel da Participação';
export const applicationMetaTitle = 'Painel da Participação no eDemocracia';
export const applicationMetaDescription = 'O painel mostra as estatísticas do resultado da participação dos cidadãos nas ferramentas Audiências Interativas e Wikilegis do portal eDemocracia, da Câmara dos Deputados. Os dados referem-se à contribuição dos cidadãos por meio de votos, opiniões, perguntas e mensagens. Visualize, baixe os dados abertos e faça suas análises comparativas.';
export const applicationProductionUrl = process.env.NEXT_PUBLIC_PARTICIPATION_SITE_URL;

// Page OG informations
export const applicationOGTitle = 'Painel da Participação no eDemocracia';
export const applicationOGDescription = 'Conheça as estatísticas do resultado da participação dos cidadãos nas ferramentas Audiências Interativas e Wikilegis do portal eDemocracia da Câmara dos Deputados. Cidadãos contribuem por meio de votos, opiniões, perguntas e mensagens. Visualize, baixe os dados abertos e faça suas análises comparativas.';
export const applicationOGSiteName = 'Painel da Participação dos Cidadãos';
