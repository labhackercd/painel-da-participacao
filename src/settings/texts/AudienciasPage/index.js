/* eslint-disable import/prefer-default-export */
import { APPLICATION_NAME } from '../../applicationOptions/index';

// Page Texts
export const pageToolTitle = 'Audiências Interativas';
export const pageHTMLTitle = `Audiências Interativas | ${APPLICATION_NAME}`;
export const pageTitle = 'A Participação em Audiências Interativas';
export const pageSubTitle = 'Painel de estatísticas de participação dos cidadãos em audiências públicas e eventos interativos da Câmara dos Deputados.';
export const pageColor = '#DA7F0B';

// ==== Totals ====
export const audiencesTotalsTexts = {
  title: 'Totais no Período',
  toolTip: null,
  toolTipAriaLabel: '',

  subSectionParticipantsTotals: {
    title: 'Participantes',
    toolTip: 'Participantes são todos os usuários que enviaram mensagens, perguntas ou votos em perguntas de outros participantes.',
    toolTipAriaLabel: 'Informação sobre o termo participantes',
  },
  subSectionAudiencesTotals: {
    title: 'Audiências Interativas',
    toolTip: 'O número refere-se a audiências canceladas e realizadas, mas também inclui outros tipos de eventos - seminários, reuniões, etc.',
    toolTipAriaLabel: 'Informação sobre o termo audiências interativas',
  },
  subSectionMessagesTotals: {
    title: 'Mensagens',
    toolTip: 'Mensagens enviadas na seção de Bate-papo.',
    toolTipAriaLabel: 'Informação sobre o termo mensagens',
  },
  subSectionQuestionsTotals: {
    title: 'Perguntas',
    toolTip: null,
    toolTipAriaLabel: null,
  },
};

// ====  Distribution of participation in the period chart section ====
export const distributionOfParticipationSectionTexts = {
  title: 'Distribuição da participação no período',
  toolTip: null,
  toolTipAriaLabel: null,
};

// ====  Distribution of participation in the period chart section ====
export const totalRoomsSectionTexts = {
  title: 'Total de audiências no período',
  toolTip: null,
  toolTipAriaLabel: null,
};

// ====  Ranking section ====
export const rankingSectionTexts = {
  title: 'Ranking das audiências',
  toolTip: 'O ranking das audiências também inclui todos os demais tipos de eventos com interatividade online, a exemplo de Seminários e Comissões Gerais.',
  toolTipAriaLabel: 'Seção Ranking das Audiências',
};

// ====  Users Section ====
export const usersSectionTexts = {
  title: 'Usuários',
  toolTip: null,
  toolTipAriaLabel: null,

  // New Users SubSection
  subSectionNewUsers: {
    title: 'Novos cadastros de usuários',
    toolTip: null,
    toolTipAriaLabel: null,
    observation: '* Em 2017 houve migração do cadastro de usuários da versão antiga para a atual. ',
    observationAnchor: `${process.env.NEXT_PUBLIC_EDEMOCRACIA_BASE_URL}/emonitor/sobre`,
    observationAnchorString: 'Mais detalhes.',
  },

  // Accumulated Total of Registered Users SubSection
  subSectionAccumulatedRegisteredUsers: {
    title: 'Total Acumulado de Usuários Cadastrados',
    toolTip: null,
    toolTipAriaLabel: null,
  },
};
