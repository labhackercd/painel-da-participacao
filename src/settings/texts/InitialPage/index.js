/* eslint-disable import/prefer-default-export */

import apiCaroulselImg from '../../../assets/images/caroulselImages/apicaroulsel.png';
import audienciasCaroulselImg from '../../../assets/images/caroulselImages/audienciascaroulsel.png';
import sobreCaroulselImg from '../../../assets/images/caroulselImages/sobrecaroulsel.png';
import wikilegisCaroulselImg from '../../../assets/images/caroulselImages/wikilegiscaroulsel.png';

export const carouselItens = [
  {
    id: 1,
    toolName: 'Audiências Interativas',
    description: 'Perguntas aos parlamentares',
    color: '#F59D2A',
    buttonText: 'Veja a participação',
    toolPage: `${process.env.NEXT_PUBLIC_AUDIENCIAS_PAGE_URL}`,
    img: audienciasCaroulselImg,
  },
  {
    id: 2,
    toolName: 'Wikilegis',
    description: 'Opiniões em textos de propostas legislativas',
    color: '#14D768',
    buttonText: 'Veja a participação',
    toolPage: `${process.env.NEXT_PUBLIC_WIKILEGIS_PAGE_URL}`,
    img: wikilegisCaroulselImg,
  },
  {
    id: 3,
    toolName: 'Sobre',
    description: 'Critérios e definições para os dados',
    color: '#1181E9',
    buttonText: 'Saiba mais',
    toolPage: `${process.env.NEXT_PUBLIC_SOBRE_PAGE_URL}`,
    img: sobreCaroulselImg,
  },
  {
    id: 4,
    toolName: 'APIs e Códigos abertos',
    description: 'Ferramentas para desenvolvedores',
    color: '#CE242E',
    buttonText: 'Acesse APIs e Códigos',
    toolPage: `${process.env.NEXT_PUBLIC_API_PAGE_URL}`,
    img: apiCaroulselImg,
  },
];
