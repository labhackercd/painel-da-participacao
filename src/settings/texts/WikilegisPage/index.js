/* eslint-disable import/prefer-default-export */
import { APPLICATION_NAME } from '../../applicationOptions/index';

// Page Texts
export const pageToolTitle = 'Wikilegis';
export const pageHTMLTitle = `Wikilegis | ${APPLICATION_NAME}`;
export const pageTitle = 'Participação na Wikilegis';
export const pageSubTitle = 'Painel de estatísticas da participação dos cidadãos nas propostas legislativas da Câmara dos Deputados.';
export const pageColor = '#00C354';

// ==== Totals ====
export const wikilegisTotalsSectionTexts = {
  title: 'Totais no Período',
  toolTip: null,
  toolTipAriaLabel: null,

  subSectionParticipantsTotals: {
    title: 'Participantes',
    toolTip: 'Participantes são todos os usuários que enviaram suas opiniões ou votaram em outras opiniões.',
    toolTipAriaLabel: 'Informação sobre o termo participantes',
  },
  subSectionLegislativeProposalsTotals: {
    title: 'Propostas Legislativas',
    toolTip: 'Propostas legislativas, disponibilizadas para receber opiniões dos cidadãos, de todos os tipos, a exemplo de Projetos de Lei, Resoluções, Atos da Mesa, etc.',
    toolTipAriaLabel: 'Informação sobre o termo propostas legislativas.',
  },
  subSectionOpinionsTotals: {
    title: 'Opiniões',
    toolTip: 'Opiniões são contribuições em trechos da proposta legislativa enviadas pelos participantes. Podem ser um simples comentário ou uma sugestão mais elaborada de reformulação do texto.',
    toolTipAriaLabel: 'Informação sobre o termo Opiniões',
  },
  subSectionVotesTotals: {
    title: 'Votos nas Opiniões',
    toolTip: 'O total mostra a soma dos votos favoráveis, neutros e contrários.',
    toolTipAriaLabel: 'Informação sobre o termo votos',
  },
};

// ====  Distribution of participation in the period chart section ====
export const distributionOfParticipationSectionTexts = {
  title: 'Distribuição da participação no período',
  toolTip: null,
  toolTipAriaLabel: null,
};

// ====  Ranking section ====
export const rankingSectionTexts = {
  title: 'Ranking das propostas legislativas',
  toolTip: 'O tempo de abertura para participação é definido pelo parlamentar. Os totais são consolidados na data de encerramento.',
  toolTipAriaLabel: 'Seção Ranking das Propostas Legislativas',
};

// ====  Users Section ====
export const usersSectionTexts = {
  title: 'Usuários',
  toolTip: null,
  toolTipAriaLabel: null,

  // New Users SubSection
  subSectionNewUsers: {
    title: 'Novos cadastros de usuários',
    toolTip: null,
    toolTipAriaLabel: null,
  },

  // Accumulated Total of Registered Users SubSection
  subSectionAccumulatedRegisteredUsers: {
    title: 'Total Acumulado de Usuários Cadastrados',
    toolTip: '',
    toolTipAriaLabel: '',
  },
};
