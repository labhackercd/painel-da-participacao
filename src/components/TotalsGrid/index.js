/* eslint-disable react/prop-types */
/* eslint-disable no-confusing-arrow */

import React, { useState } from 'react';
import PropTypes from 'prop-types';
import CircularProgress from '@material-ui/core/CircularProgress';
import {
  Box, Typography,
} from '@material-ui/core';
import CheckCircleIcon from '@material-ui/icons/CheckCircle';
import Tooltips from '../ToolTip/index';
import InfoModal from '../InfoModal';
import { useStyles } from './style';
import useMobile from '../../hooks/useMobile';

function ToolTip(props) {
  const {
    // eslint-disable-next-line react/prop-types
    itens, index,
  } = props;
  const [open, setOpen] = useState(false);
  const { mobileView, tabletView } = useMobile();
  const handleOpen = () => (mobileView || tabletView) && setOpen(true);
  const handleClose = () => setOpen(false);

  return (
    <>
      {(itens[index].toolTipText !== null && itens[index].toolTipText !== undefined)
      && (
        <>
          <Box onClick={handleOpen} flexGrow={1} display="flex" alignItems="center" style={{ margin: 0 }}>
            <Tooltips
              toolTipText={itens[index].toolTipText}
              toolTipColor={itens[index].toolTipColor}
              toolTipAriaLabel={itens[index].toolTipAriaLabel}
            />
          </Box>
          <InfoModal
            open={open}
            handleClose={handleClose}
            title={itens[index].title}
            toolTipText={itens[index].toolTipText}
            toolTipColor={itens[index].toolTipColor}
          />
        </>
      )}
    </>
  );
}

export default function TotalsGrid(props) {
  const classes = useStyles();
  const {
    itens, className,
  } = props;

  return (
    <Box className={`${classes.root} ${className}`}>
      {itens.map((item, index) => (
        <>
          {item.isLoaded ? (
            <>
              <Box key={item.title} display="flex" flexDirection="column" flex={1} flexBasis="45%">
                <Box display="flex">
                  <Typography variant="h5" className={classes.text}>
                    <Box display="flex">
                      {item.title}
                    </Box>
                  </Typography>
                  <ToolTip itens={itens} index={index} />
                </Box>
                <Box paddingBottom={(index === 2 || index === 3) ? '16px' : '0px'}>
                  <Typography className={classes.info} noWrap>
                    {item.info}
                  </Typography>
                  { item.subInformation && item.subInformation.length !== 0 && (
                    <Box width="100%" display="flex" alignItems="center" paddingLeft="16px" paddingTop="8px" paddingBottom="8px">
                      <Box marginRight="10px">
                        <CheckCircleIcon fontSize="small" style={{ color: item.toolTipColor }} />
                      </Box>
                      <Box display="flex" alignItems="center" marginBottom="4px">
                        <Typography style={{ color: '#FFF', fontSize: '13px' }} noWrap>
                          {item.subInformation}
                        </Typography>
                      </Box>
                    </Box>
                  )}
                </Box>
              </Box>
              { index % 2 === 0 && (
                <hr flexBasis="10%" flexShrink={0} color="#363636" />
              )}
              { index === 1 && (
                <hr flexBasis="100%" flexShrink={0} color="#363636" width="100%" height="1px" />
              )}
            </>
          ) : (
            <Box display="flex" alignItems="center" justifyContent="center" width="100%" height="100%">
              <CircularProgress />
            </Box>
          )}
        </>
      ))}
    </Box>
  );
}

TotalsGrid.propTypes = {
  itens: PropTypes.array,
  className: PropTypes.object,
};

TotalsGrid.defaultProps = {
  itens: [],
  className: {},
};
