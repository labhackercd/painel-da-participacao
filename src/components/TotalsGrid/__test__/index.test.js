import React from 'react';
import { shallow, mount } from 'enzyme';
import TotalsGrid, { ToolTip } from '../index';
import MockTheme from '../../../mocks/theme/mockTheme';
import * as TEXTCONSTANTS from '../../../settings/texts/AudienciasPage';

const mockData = [
  {
    isLoaded: true,
    info: '5000',
    title: TEXTCONSTANTS.audiencesTotalsTexts.subSectionParticipantsTotals.title,
    toolTipText: TEXTCONSTANTS.audiencesTotalsTexts.subSectionParticipantsTotals.toolTip,
    toolTipAriaLabel:
      TEXTCONSTANTS.audiencesTotalsTexts.subSectionParticipantsTotals.toolTipAriaLabel,
    toolTipColor: '#DA7F0B',
  },
  {
    isLoaded: true,
    info: '5000',
    title: TEXTCONSTANTS.audiencesTotalsTexts.subSectionAudiencesTotals.title,
    toolTipText: TEXTCONSTANTS.audiencesTotalsTexts.subSectionAudiencesTotals.toolTip,
    toolTipAriaLabel: TEXTCONSTANTS.audiencesTotalsTexts.subSectionAudiencesTotals.toolTipAriaLabel,
    toolTipColor: '#DA7F0B',
    subInformation: '200 realizadas',
  },
  {
    isLoaded: true,
    info: '5000',
    title: TEXTCONSTANTS.audiencesTotalsTexts.subSectionMessagesTotals.title,
    toolTipText: TEXTCONSTANTS.audiencesTotalsTexts.subSectionMessagesTotals.toolTip,
    toolTipAriaLabel: TEXTCONSTANTS.audiencesTotalsTexts.subSectionMessagesTotals.toolTipAriaLabel,
    toolTipColor: '#DA7F0B',
  },
  {
    isLoaded: true,
    info: '5000',
    title: TEXTCONSTANTS.audiencesTotalsTexts.subSectionQuestionsTotals.title,
    toolTipText: TEXTCONSTANTS.audiencesTotalsTexts.subSectionQuestionsTotals.toolTip,
    toolTipAriaLabel: TEXTCONSTANTS.audiencesTotalsTexts.subSectionQuestionsTotals.toolTipAriaLabel,
  },
];

it('snapshot should not have changes', () => {
  const component = shallow(<MockTheme><TotalsGrid itens={[]} /></MockTheme>);
  expect(component.exists()).toEqual(true);
  expect(component).toMatchSnapshot();
});

it('Render Totals Grid With Data', () => {
  const component = mount(<MockTheme><TotalsGrid itens={mockData} /></MockTheme>);
  expect(component.exists()).toEqual(true);
  expect(component).toMatchSnapshot();
});

it('snapshot should not have changes', () => {
  const component = shallow(<MockTheme><ToolTip itens={[]} /></MockTheme>);
  expect(component.exists()).toEqual(true);
  expect(component).toMatchSnapshot();
});

it('Render Totals Grid ToolTip With Data', () => {
  const component = mount(<MockTheme><ToolTip itens={mockData} /></MockTheme>);
  expect(component.exists()).toEqual(true);
  expect(component).toMatchSnapshot();
});
