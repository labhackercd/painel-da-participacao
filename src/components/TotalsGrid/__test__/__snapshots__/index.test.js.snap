// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Totals Grid ToolTip With Data 1`] = `ReactWrapper {}`;

exports[`Render Totals Grid With Data 1`] = `ReactWrapper {}`;

exports[`snapshot should not have changes 1`] = `ShallowWrapper {}`;

exports[`snapshot should not have changes 2`] = `ShallowWrapper {}`;
